README.md

```diff
@@ -3,63 +3,116 @@
+- [virtualenv](https://virtualenv.pypa.io)
+- [pip](https://pip.pypa.io/)
+
+### Create a TwiML App
+
+This project is configured to use a **TwiML App**, which allows us to easily set the voice URLs for all Twilio phone numbers we purchase in this app.
+
+To create a new TwiML app click [here](https://www.twilio.com/console/voice/twiml/apps).
+
+![](images/create-twiml-app.png)
+
+### Install Add-ons
+
+The following guide will help you to [install Add-ons](https://www.twilio.com/docs/add-ons/install). You can access the Add-ons in the Twilio console [here](https://www.twilio.com/console/add-ons). The Spam Filtering Add-ons that are used on this application are:
+- [Ekata Phone Validation](https://showcase.twilio.com/s/partner-listing/a8E1W00000097QEUAY)
+- [Marchex Clean Call](https://showcase.twilio.com/s/partner-listing/a8E1W00000097QxUAI)
+- [Nomorobo Spam Score](https://showcase.twilio.com/s/partner-listing/a8E1W00000097R7UAI)
+
+Once you've selected the Add-on, just click on `Install` button. Then, you will see a pop-up window where you should read and agree the terms, then, click the button `Agree & Install`. For this application, you just need to handle the incoming voice calls, so make sure the `Incoming Voice Call` box for `Use In` is checked and click `Save`
+
+![](images/install-add-on.png)
+
+### Local Development
+
+1. First clone this repository and `cd` into it:
+
+   ```bash
    git clone https://github.com/TwilioDevEd/block-spam-calls-python.git
    cd block-spam-calls-python
    ```
 
-1. Create a virtualenv and load it
+1. Create the virtualenv, load it and install dependencies:
 
-    ```
-    virtualenv venv
-    source ./venv/bin/activate
+    ```bash
+    make install
     ```
 
-1. Install dependencies:
+1. Run the application (will run on port 5000).
 
-    ```
-    pip install -r requirements.txt
+    ```bash
+    make serve
     ```
 
-1. Run the application.
+1. To actually forward incoming calls, your development server will need to be publicly accessible. [We recommend using ngrok to solve this problem](https://www.twilio.com/blog/2015/09/6-awesome-reasons-to-use-ngrok-when-testing-webhooks.html).
 
-    ```
-    python block_spam_calls/app.py
+    ```bash
+    ngrok http 5000
     ```
 
-To actually forward incoming calls, your development server will need to be publicly accessible. [We recommend using ngrok to solve this problem](https://www.twilio.com/blog/2015/09/6-awesome-reasons-to-use-ngrok-when-testing-webhooks.html).
+1. Once you have started ngrok, update your [TwiML app's](https://www.twilio.com/console/voice/twiml/apps) Voice URL setting to use your ngrok hostname, so it will look something like this:
 
-Once you have started ngrok, update your TwiML app's voice URL setting to use your ngrok hostname, so it will look something like this:
+    ```bash
+    http://88b37ada.ngrok.io/
+    ```
 
-```
-http://88b37ada.ngrok.io/
-```
+That's it!
```

Comment:
we should provide more instructions on how this works and how to test it

Reply:
additionally, this isn't working for me. im getting this: 
```  File "/Users/bmurali/Developer/ce-apps/block-spam-calls-python/block_spam_calls/app.py", line 24, in block_spam_calls
    should_be_blocked = blocker_func(add_ons['results'].get(blocker_name, {}))
  File "/Users/bmurali/Developer/ce-apps/block-spam-calls-python/block_spam_calls/app.py", line 47, in should_be_blocked_by_ekata
    return ekata['result']['reputation_level'] >= 4
KeyError: 'reputation_level'```

i wonder if the add-on changed their API :(

Reply:
Hey Bharat, sorry about that! Could you please help me with this one? I've changed Whitepages to Ekata, but just changed the name of the add-on, then I executed the tests. Also, I've tested the POST method with a REST client (Postman). Could you help me with some steps to test the app behavior without a REST client? Thanks!

Reply:
the app seems to be working, so I think we are ok for now. I would add one more line to the final step staying "click on `Call` to test your app"